File: jscomp/test/module_parameter_test.js
'use strict';

var Mt = require("./mt.js");
var $$String = require("../../lib/js/string.js");

function u(v) {
  return v;
}

var s = $$String;

var N = {
  s: s
};

function v(x) {
  return x.length;
}

var suites_0 = [
  "const",
  (function (param) {
      return {
              tag: /* Eq */0,
              _0: 1,
              _1: 1
            };
    })
];

var suites_1 = /* :: */{
  _0: [
    "other",
    (function (param) {
        return {
                tag: /* Eq */0,
                _0: 3,
                _1: 3
              };
      })
  ],
  _1: /* [] */0
};

var suites = /* :: */{
  _0: suites_0,
  _1: suites_1
};

Mt.from_pair_suites("Module_parameter_test", suites);

var v0 = 1;

exports.u = u;
exports.N = N;
exports.v0 = v0;
exports.v = v;
exports.suites = suites;
/*  Not a pure module */

File: jscomp/test/module_parameter_test.ml
module type X = module type of String


let u (v : (module X)) = v

module N = 
struct 
  let s =  u (module String)
end

let v0 = 
  let module V = (val N.s : X ) in V.length "x" 

let v x = 
  let module V = (val N.s : X ) in V.length x


let suites = Mt.[
    "const", (fun _ -> Eq(1,v0));
    "other", (fun _ -> Eq(3,v "abc"))     
]

;; Mt.from_pair_suites __MODULE__ suites
